src/components/trade/TradesTable/index.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { useState } from 'react'
+import styled from 'styled-components'
+import BigNumber from 'bignumber.js'
+import { Link } from 'react-router-dom'
+import { faExchangeAlt } from '@fortawesome/free-solid-svg-icons'
+
+import { TokenErc20 } from '@gnosis.pm/dex-js'
+import { Trade, RawOrder } from 'api/operator'
+
+import { DateDisplay } from 'components/common/DateDisplay'
+import { RowWithCopyButton } from 'components/common/RowWithCopyButton'
+import {
+  formatSmartMaxPrecision,
+  getOrderLimitPrice,
+  getOrderExecutedPrice,
+  formatCalculatedPriceToDisplay,
+  formatExecutedPriceToDisplay,
+} from 'utils'
+import { HelpTooltip } from 'components/Tooltip'
+import StyledUserDetailsTable, {
+  StyledUserDetailsTableProps,
+  EmptyItemWrapper,
+} from '../../common/StyledUserDetailsTable'
+import Icon from 'components/Icon'
+import TradeOrderType from 'components/common/TradeOrderType'
+import { Surplus } from './Surplus'
+import { RAW_ORDER } from '../../../../test/data'
+
+const Wrapper = styled(StyledUserDetailsTable)`
+  > thead > tr,
+  > tbody > tr {
+    grid-template-columns: 10rem 4rem repeat(2, 10rem) repeat(2, 14rem) 10rem 10rem 1fr;
+  }
+`
+
+const TxHash = styled.div`
+  max-width: 10rem;
+  overflow: hidden;
+  text-overflow: ellipsis;
+`
+
+function isTokenErc20(token: TokenErc20 | null | undefined): token is TokenErc20 {
+  return (token as TokenErc20).address !== undefined
+}
+
+function formattedAmount(erc20: TokenErc20 | null | undefined, amount: BigNumber): string {
```

Comment:
These helper fns are pretty the same as https://github.com/gnosis/gp-ui/blob/1dfe1e7702d505c64d62460b7f1574d40d47124e/src/components/orders/OrdersUserDetailsTable/index.tsx#L29 Henry created, right?

Could you move them to a common utils file?